test/Microsoft.Azure.SignalR.Tests/AddAzureSignalRFacts.cs

```diff
@@ -85,8 +89,41 @@ public void AddAzureReadsConnectionStringFirst()
 
             var options = serviceProvider.GetRequiredService<IOptions<ServiceOptions>>().Value;
 
-            Assert.Equal("myconnectionstring", options.ConnectionString);
-            Assert.Equal("myconnectionstring", capturedConnectionString);
+            Assert.Equal(DefaultValue, options.ConnectionString);
+            Assert.Equal(DefaultValue, capturedConnectionString);
+        }
+
+        [Theory]
+        [InlineData(CustomValue, null, null, CustomValue)]
+        [InlineData(CustomValue, DefaultValue, SecondaryValue, CustomValue)]
+        [InlineData(null, DefaultValue, SecondaryValue, DefaultValue)]
+        [InlineData(null, null, SecondaryValue, SecondaryValue)]
+        public void AddAzureSignalRLoadConnectionStringOrder(string customValue, string defaultValue,
+            string secondaryValue, string expected)
+        {
+            var services = new ServiceCollection();
+            var config = new ConfigurationBuilder()
+                            .AddInMemoryCollection(new Dictionary<string, string>
+                            {
+                                {"Azure:SignalR:ConnectionString", defaultValue},
+                                {"ConnectionStrings:Azure:SignalR:ConnectionString", secondaryValue}
```

Comment:
This looks so strange 😄. `ConnectionStrings:Azure:SignalR:ConnectionString`

Should it be `ConnectionStrings:Azure:SignalR` instead?

Reply:
Are ConnectionStrings section mainly for SQL databases? When I create a connection string in Azure portal, I need to choose from MySQL, SQLServer, SQLAzure, and Custom. What's the difference? Should user choose Custom when using SignalR?

Reply:
Usually it's a database connection string. For example, I don't think you specify your service bus connection string in there.

Reply:
Agree that the key is repetitive and doesn't sound natural.
However this fix is specifically for Azure Web Apps. When user "accidentally" put the connection string in the `Connection Strings` section, we can still pick it up. So if we change the key name, it means we provide another key and need to document it.
IMO, this is not a recommended approach to set the connection string in environment. But if user prefer to put it in `Connection Strings` section, the Service SDK can still work.